detect.go

```diff
@@ -0,0 +1,67 @@
+package composer
+
+import (
+	"github.com/paketo-buildpacks/packit/v2"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+	"os"
+	"path/filepath"
+)
+
+const DefaultComposerPath = "composer.json"
+
+func Detect() packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		composerPath, composerVarFound := os.LookupEnv("COMPOSER")
+		if !composerVarFound {
+			composerPath = DefaultComposerPath
+		}
+
+		if exists, err := fs.Exists(filepath.Join(context.WorkingDir, composerPath)); err != nil {
+			return packit.DetectResult{}, err
+		} else if !exists && !composerVarFound {
+			return packit.DetectResult{}, packit.Fail.WithMessage("no %s found", DefaultComposerPath)
+		} else if !exists && composerVarFound {
+			return packit.DetectResult{}, packit.Fail.WithMessage("no %s found at location '%s'", DefaultComposerPath, composerPath)
+		}
+
+		composerRequirement := packit.BuildPlanRequirement{
+			Name: "composer",
+			Metadata: BuildPlanMetadata{
+				Build: true,
+			},
+		}
+
+		if version, versionOk := os.LookupEnv("BP_COMPOSER_VERSION"); versionOk {
+			composerRequirement.Metadata = BuildPlanMetadata{
+				VersionSource: "BP_COMPOSER_VERSION",
+				Version:       version,
+				Build:         true,
+			}
+		}
```

Comment:
I think since the composer buildpack is looking at this variable already, there's no need to double check it here in this buildpack!